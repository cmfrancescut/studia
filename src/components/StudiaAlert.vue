<script setup>
import { XMarkIcon } from '@heroicons/vue/24/solid'
import { ref, computed } from 'vue'

const props = defineProps({
  variant: {
    type: String,
    validator(value) {
      return ['info', 'warning', 'danger', 'success'].includes(value)
    }
  },
  dismissable: {
    type: Boolean,
    default: false
  }
})

const visible = ref(true)
const style = computed(() => {
  switch (props.variant) {
    case 'info':
      return 'bg-info-100 border-info-500 text-info-700'
    case 'warning':
      return 'bg-warning-100 border-warning-500 text-warning-700'
    case 'danger':
      return 'bg-danger-100 border-danger-500 text-danger-700'
    case 'success':
      return 'bg-success-100 border-success-500 text-success-700'
    default:
      return 'bg-gray-50 text-gray-700'
  }
})

function toggleVisibility() {
  visible.value = !visible.value
}
</script>
<style scoped>
@import '../style.css';
</style>
<template>
  <div v-if="visible">
    <div class="flex flex-col p-4 m-4 gap-y-2 border border-1 rounded" :class="style">
      <div class="flex flex-row">
        <div class="font-bold uppercase w-[99%] max-h-6">
          <slot name="title"></slot>
        </div>
        <div v-if="props.dismissable">
          <button @click="toggleVisibility">
            <XMarkIcon class="h-6 w-6 pl-0" />
          </button>
        </div>
      </div>
      <div>
        <slot></slot>
      </div>
    </div>
  </div>
</template>
