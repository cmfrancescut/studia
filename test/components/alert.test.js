import { mount } from '@vue/test-utils'
import StudiaAlert from "@/components/StudiaAlert.vue";

test('Mount component', () => {
  expect(StudiaAlert).toBeTruthy()
  const wrapper = mount(StudiaAlert, {slots: { 'default': 'text', 'title': 'heading'}})

  expect(wrapper.text()).toContain('text')
  expect(wrapper.text()).toContain('heading')
  expect(wrapper.html()).toMatchSnapshot()
})

