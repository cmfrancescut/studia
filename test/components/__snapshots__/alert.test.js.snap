// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-030bca60=\\"\\">
  <div data-v-030bca60=\\"\\" class=\\"flex flex-col p-4 m-4 gap-y-2 border border-1 rounded bg-gray-50 text-gray-700\\">
    <div data-v-030bca60=\\"\\" class=\\"flex flex-row\\">
      <div data-v-030bca60=\\"\\" class=\\"font-bold uppercase w-[99%] max-h-6\\">heading</div>
      <!--v-if-->
    </div>
    <div data-v-030bca60=\\"\\">text</div>
  </div>
</div>"
`;
